File: Array/DecompressRLE.cpp
#include<vector>

using namespace std;

class DecompressRLE {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int> output;
     for (int index = 0; index < nums.size(); ++index){
         int freq = nums[index];
         int val = nums[index + 1];
         while(freq > 0) {
             output.push_back(val);
             --freq;
         }
         ++index;
     }
        return output;
    }
};

File: Array/DecompressRLE/go/DecompressRLE.go
package _go

// deCompressRLE : iterate through the freq and append to array
// Time: O(n)
// Space: O(n)
func decompressRLElist(nums []int) []int {
	res := make([]int, 0)
	for i := 0; i < len(nums); i += 2 {
		key, value := nums[i], nums[i+1]
		for j := 0; j < key; j++ {
			res = append(res, value)
		}
	}
	return res
}
